Tests/test_units.py

```diff
@@ -0,0 +1,44 @@
+
+# Copyright (c) 2020, Build-A-Cell. All rights reserved.
+# See LICENSE file in the project root directory for details.
+
+from unittest import TestCase
+from biocrnpyler.sbmlutil import create_sbml_model
+import biocrnpyler.units as units
+import pytest
+import libsbml
+
+def check(value, message):
+        """If 'value' is None, prints an error message constructed using
+        'message' and then exits with status code 1 (for libsbml). If 'value' is an integer,
+        it assumes it is a libSBML return status code.  If the code value is
+        LIBSBML_OPERATION_SUCCESS, returns without further action; if it is not,
+        prints an error message constructed using 'message' along with text from
+        libSBML explaining the meaning of the code, and exits with status code 1.
+        """
+        if value == None:
+            raise SystemExit(
+                'LibSBML returned a null value trying to ' + message + '.')
+        elif type(value) is int:
+            if value == libsbml.LIBSBML_OPERATION_SUCCESS:
+                return
+            else:
+                err_msg = 'Error encountered trying to ' + message + '.' \
+                    + 'LibSBML returned error code ' + str(value) + ': "' \
+                    + libsbml.OperationReturnValue_toString(value).strip() + '"'
+                raise SystemExit(err_msg)
+        else:
+            return
+
+class TestSpecies(TestCase):
+    def test_units_initialization(self):
+        # tests naming convention repr without species type or attributes
+        document, sbml_model = create_sbml_model()
+        list_of_supported_units = ["nL", "uL", "mL", "L",
```

Comment:
Not sure how hard this would be, but could we parse things like nL/s to nL_per_second and automatically generate some of these combinations from the primitives?

Reply:
Sure, we could do that..Only a matter of writing some string parsing code. But automatically writing the unit definitions could be tricky from the string letters such as "n" in "nL" and "m" in "mL/s"  (you need that for 10^-9  or 10^-6) etc or even it could get complicated like "nanomole_per_litre_per_second" or other stuff.

I think creating new unit definitions manually is very easy since you just have to copy a function in units.py and create your own by changing the exponents of different things and using your own string identifiers.